crates/iceberg/src/metadata_scan.rs

```diff
@@ -95,7 +99,17 @@ impl<'a> SnapshotsTable<'a> {
     }
 
     /// Scans the snapshots table.
-    pub fn scan(&self) -> Result<RecordBatch> {
+    pub async fn scan(&self) -> Result<ArrowRecordBatchStream> {
+        let arrow_schema = Arc::new(self.schema());
+        let table_metadata = self.table.metadata_ref();
+
+        Ok(
+            futures::stream::once(async move { Self::build_batch(arrow_schema, &table_metadata) })
+                .boxed(),
+        )
+    }
+
+    fn build_batch(arrow_schema: SchemaRef, table_metadata: &TableMetadata) -> Result<RecordBatch> {
```

Comment:
Perhaps we could simply implement it in `scan` instead.

Reply:
Moved into `scan()`.

Reply:
(Did the same for the manifests table as well.)